vsc_pdf_template_transformer/lib/utils/widget_json_converter.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:vsc_pdf_template_transformer/models/tpl_string.dart';
+import 'package:vsc_pdf_template_transformer/models/tpl_text.dart';
+import '../utils/widget_builder.dart' as wb;
+
+class WidgetJsonConverter
+    extends JsonConverter<wb.WidgetBuilder?, Map<String, dynamic>?> {
+  const WidgetJsonConverter();
+
+  @override
+  wb.WidgetBuilder? fromJson(Map<String, dynamic>? json) {
+    if (json != null) {
+      final res = TplText.fromJson(json);
+      return res;
+    }
+    return null;
+  }
+
+  @override
+  Map<String, dynamic>? toJson(wb.WidgetBuilder? object) {
+    if (object != null && object is TplText) {
```

Comment:
If you included:
```
Map<String, dynamic> toJson();
```
on WidgetBuilder, then this just becomes `object?.toJson()` and you won't need conditional logic for each widget type.

Also, `TplText(text: TplString(value: object.text.value)).toJson()`  would seem to lose `object.text.expression` if there was one.